### PR TITLE
Rework EE Replacement of Vanilla materials

### LINKED ISSUE #256: [DISABLED ITEM] Raw Pendorite
### Modpack Version

1.5.0

### Describe your issue.

I just got raw pendorite from mining it in the nether.

image
image

### Crash Report

No crash

### Latest Log

Not needed

### Have you modified the modpack?

No

### User Modifications

Nope, default settings

### Did the issue happen in singleplayer or on a server?

Singleplayer

### Discord Username

Juknum#0021

### LINKED ISSUE #336: Netherite tools and armor require minecraft netherite in smithing table
### Modpack Version

1.6.0

### Describe your issue.

cant upgrade vanilla tools and armor with EE netherite, it requires vanilla netherite.

### Crash Report

_No response_

### Latest Log

_No response_

### Have you modified the modpack?

No

### User Modifications

_No response_

### Did the issue happen in singleplayer or on a server?

Both

### Discord Username

_No response_

### LINKED ISSUE #349: Hexerei Stripped Logs Missing Forge Tag
### Modpack Version

1.6.0

### Describe your issue.

Hexerei stripped logs(willow, witch hazel and mohagany) are missing the `forge:stripped_logs` tag making them unuseable for recipes requiring stripped logs.

### Crash Report

_No response_

### Latest Log

_No response_

### Have you modified the modpack?

No

### User Modifications

_No response_

### Did the issue happen in singleplayer or on a server?

Both

### Discord Username

_No response_

### LINKED ISSUE #351: No osmium
### Modpack Version

no idea

### Describe your issue.

Unable to locate osmium ore in several hours mining, game un-playable

### Crash Report

_No response_

### Latest Log

_No response_

### Have you modified the modpack?

No

### User Modifications

_No response_

### Did the issue happen in singleplayer or on a server?

Server

### Discord Username

merchantofdeath#2094

### LINKED ISSUE #357: Create Alternator Recipe Problem
### Modpack Version

1.6.0

### Describe your issue.

When trying to craft the Create Crafts and addons alternator it needs the create specific andesite alloy and not the EE version.

### Crash Report

_No response_

### Latest Log

_No response_

### Have you modified the modpack?

Yes

### User Modifications

CCI

### Did the issue happen in singleplayer or on a server?

Singleplayer

### Discord Username

JoprebondOdin#5307

### PR DESCRIPTION
**Pull request by Kanzaji**
# Current Changes
- [x] Fixed scripts that used EE variants (Except expert ones, those I leave up to Motoko because I will mess them up 😅)
- [x] Improved error handling in Ore Processing Unification, added Raw Block to Arc Furnace compat
- [x] Added Exception Support for Loot Unification
- [x] Remove vanilla items from EE, just replace Assets
- [x] Add important note to EE materials, that they are going to be gone next breaking update (or on release version of the EE update)
- [x] Add conversion recipes for EE stuff
- [x] ~~Re-Enabled default Blue Skies ore gen, until EE one is ready~~
- [x] ~~Changed from EE materials listed above to original items, due to AU issues that are WIP and Hard-Coded content >.>~~
- [x] Resolves #351
- [x] Resolves #349 
- [x] Resolves #256 and #354 (Yes, I found out what was wrong and why it was actually still generating >.>)
- [x] Resolves #336
- [x] Re-Balance Nether Ore Generation after resolving #256 
- [x] Update quests to include changes made by this PR and removal of the blue skies
- [x] Resolves #357 

EE Materials are left until next breaking release, and for now are using modified texture to show that they are going to be removed:
obraz

Conversion recipes for all materials from screenshot above:
obraz

Added Osmium generation in nether!

## Items designated to removal in another release / breaking update from EE:
- All Blue Skies material related stuff
- Copper: Ingot / Storage Block / Raw + Raw Block
- Iron: Ingot / Nugget / Storage Block / Raw + Raw Block
- Gold: Ingot / Nugget / Storage Block / Raw + Raw Block
- Diamond: Storage Block
- Emerald: Storage Block
- Netherite: Ingot / Storage Block (iirc nuggets aren't vanilla 😅)

### Files to delete next breaking update:
- `config\emendatusenigmatica\strata\crystal_sandstone.json`
- `config\emendatusenigmatica\strata\lunar_cobblestone.json`
- `config\emendatusenigmatica\strata\lunar_stone.json`
- `config\emendatusenigmatica\strata\midnight_sandstone.json`
- `config\emendatusenigmatica\strata\mossy_lunar_cobblestone.json`
- `config\emendatusenigmatica\strata\rimestone.json`
- `config\emendatusenigmatica\strata\taratite.json`
- `config\emendatusenigmatica\strata\turquoise_stone.json`
- `config\emendatusenigmatica\strata\umber.json`
- `config\emendatusenigmatica\material\aquite.json`
- `config\emendatusenigmatica\material\charoite.json`
- `config\emendatusenigmatica\material\diopside.json`
- `config\emendatusenigmatica\material\falsite.json`
- `config\emendatusenigmatica\material\horizonite.json`
- `config\emendatusenigmatica\material\pyrope.json`
- `config\emendatusenigmatica\material\ventium.json`
- `kubejs\client_scripts\base\jei_conversion_note.js`
- `kubejs\server_scripts\base\recipes\emendatusenigmatica\EE_conversion.js`
- All KubeJS scripts with note TODO: Delete next breaking update
- Blue Skies Assets are already deleted, and they technically shouldn't cause any issues if present, so I'm just mentioning them here.

### CODE REVIEW

File: kubejs/server_scripts/expert/recipes/enigmatica/dust_crushing_tiers.js [hunk 1]
```diff
@@ -46,6 +46,8 @@ ServerEvents.recipes((event) => {
 
     const metals = Object.keys(metal_properties);
 
+    // Note to Motoko
+    // Due to issues with AU and hard-coded content, vanilla ingots are now from Minecraft! This script requires a bit of a rework to add support for minecraft namespace -- Kanz
```
Comment (DAmNRelentless) [inline kubejs/server_scripts/expert/recipes/enigmatica/dust_crushing_tiers.js:50]: What is this about? Which issues and what hard-coded content?

Comment (MuteTiefling) [reply kubejs/server_scripts/expert/recipes/enigmatica/dust_crushing_tiers.js:50]: Anvil repairs mostly.

Comment (DAmNRelentless) [reply kubejs/server_scripts/expert/recipes/enigmatica/dust_crushing_tiers.js:50]: Yea, non-JSON-based recipes can't be covered by AU. The other issues, like smithing-like recipes, should be resolved. If you have further issues, feel free to talk to us.

File: kubejs/server_scripts/expert/recipes/create/sequenced_assembly.js
Comment (MuteTiefling) [inline kubejs/server_scripts/expert/recipes/create/sequenced_assembly.js:1]: This is fixed in #332

File: kubejs/server_scripts/expert/recipes/enigmatica/dust_crushing_tiers.js
Comment (MuteTiefling) [inline kubejs/server_scripts/expert/recipes/enigmatica/dust_crushing_tiers.js:1]: This is fixed in https://github.com/EnigmaticaModpacks/Enigmatica9/pull/332

**Comment by Kanzaji** [comment]
Okay I would say this is ready :D
Nether Ore Gen rebalance still requires a in-game review from anyone else than me 😅 But otherwise PR is ready to merge.

**Comment by Kanzaji** [comment]
Okay I would say this is ready :D
Nether Ore Gen rebalance still requires a in-game review from anyone else than me 😅 But otherwise PR is ready to merge.

File: kubejs/client_scripts/base/jei_ee_conversion_note.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+// https://github.com/EnigmaticaModpacks/Enigmatica9/pull/338
+// It is used to add conversion note for all items that are going to be removed!
+
+JEIEvents.information((event) => {
+    const materials = ["copper", "iron", "gold", "diamond", "emerald", "netherite"]
+    const types = ["ingot", "nugget", "block"]
+    const conversion_message = [
+        Text.of("This item is going to be removed in the coming updates!").red(),
+        " ",
+        Text.of("Please convert it to vanilla variant before updating your instance.").red()
```
Comment (NielsPilgaard) [inline kubejs/client_scripts/base/jei_ee_conversion_note.js:13]:
```suggestion
        Text.of("Please convert it to the vanilla variant before updating.").red()
```

File: kubejs/client_scripts/base/jei_ee_conversion_note.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+// This script is TEMPORARY!
+// It should be removed in next release / breaking update, with additional removal of EE Vanilla materials, full list of items removed can be found here:
+// https://github.com/EnigmaticaModpacks/Enigmatica9/pull/338
+// It is used to add conversion note for all items that are going to be removed!
+
+JEIEvents.information((event) => {
+    const materials = ["copper", "iron", "gold", "diamond", "emerald", "netherite"]
+    const types = ["ingot", "nugget", "block"]
+    const conversion_message = [
+        Text.of("This item is going to be removed in the coming updates!").red(),
```
Comment (NielsPilgaard) [inline kubejs/client_scripts/base/jei_ee_conversion_note.js:11]:
```suggestion
        Text.of("This item is going to be removed in the near future.").red(),
```

File: kubejs/server_scripts/base/entity_events/spawned/custom_rituals.js [hunk 4]
```diff
@@ -198,6 +198,7 @@ EntityEvents.spawned((event) => {
                         schedule.server.runCommandSilent(command);
                     }
 
+                    // Note to Motoko: Due to pending removal of Blue_Skies, this require a change I can't do myself ^^"
```
Comment (NielsPilgaard) [inline kubejs/server_scripts/base/entity_events/spawned/custom_rituals.js:201]: @MuteTiefling heads up 😄

File: kubejs/server_scripts/base/entity_events/spawned/custom_rituals.js [hunk 5]
```diff
@@ -383,6 +384,7 @@ EntityEvents.spawned((event) => {
 
                 delay = start_delay + 0.5 * block.pos[1];
                 event.server.scheduleInTicks(delay, (schedule) => {
+                // Note to Motoko: Due to pending removal of Blue_Skies, this require a change I can't do myself ^^"
```
Comment (NielsPilgaard) [inline kubejs/server_scripts/base/entity_events/spawned/custom_rituals.js:387]: @MuteTiefling heads up 😄

File: kubejs/server_scripts/base/tags/items/emendatusenigmatica/gold.js [hunk 6]
```diff
@@ -1,3 +1,10 @@
 ServerEvents.tags('item', (event) => {
```
Comment (NielsPilgaard) [inline kubejs/server_scripts/base/tags/items/emendatusenigmatica/gold.js:1]: This file should be in `base/tags/items/minecraft/piglin_loved.js`

File: kubejs/server_scripts/constants/ritual_effects.js [hunk 7]
```diff
@@ -34,6 +34,7 @@ const ritual_effects = {
         }
     },
     teleport_nether: {
+        // Note to Motoko: Due to pending removal of Blue_Skies, this require a change I can't do myself ^^"
```
Comment (NielsPilgaard) [inline kubejs/server_scripts/constants/ritual_effects.js:37]: @MuteTiefling heads up 😄

File: kubejs/server_scripts/normal/unification/unify_ore_processing.js [hunk 8]
```diff
@@ -186,79 +196,139 @@ function metal_ore_processing(material, properties, event, id_prefix) {
     // Fallback output checks:
     if (crushed_ore_itemStack.isEmpty()) {
         crushed_ore_itemStack = Item.of(Ingredient.of(`#create:crushed_ores/${material}`).getItemIds()[0]);
-        if (localDebug)
-            console.warn(
-                ' // Ore Processing Rework // Material "' +
-                    material +
-                    '" uses fallback output item (Crushed Ore) for metal processing!'
-            );
+        crushed_ore_itemStack = (crushed_ore_itemStack.isEmpty() || crushed_ore_itemStack.getId() == "minecraft:barrier")? null: crushed_ore_itemStack;
```
Comment (NielsPilgaard) [inline kubejs/server_scripts/normal/unification/unify_ore_processing.js:199]: formatting looks off

**Review by NielsPilgaard — CHANGES_REQUESTED** [review]

File: kubejs/server_scripts/base/tags/items/emendatusenigmatica/gold.js (hunk 6 above)
Comment (Kanzaji) [reply kubejs/server_scripts/base/tags/items/emendatusenigmatica/gold.js:1]: Why did I name it gold in the first place 🤔

File: kubejs/server_scripts/base/entity_events/spawned/custom_rituals.js (hunk 5 above)
Comment (Kanzaji) [reply kubejs/server_scripts/base/entity_events/spawned/custom_rituals.js:387]: I think Motoko already fixed that 😅 Need to check, if so, I'm gonna remove the notes :D

**Comment by Kanzaji** [comment]
Alrighty, #357 has been fixed 😄 and notes have been read and mistakes corrected!

File: kubejs/server_scripts/base/tags/items/minecraft/piglin_loved.js
Comment (NielsPilgaard) [inline kubejs/server_scripts/base/tags/items/minecraft/piglin_loved.js:1]: Oh I just realized, the old file needs a "delete later" note, it won't be deleted for clients updating ^^'

File: kubejs/server_scripts/normal/unification/unify_ore_processing.js [hunk 9]
```diff
@@ -196,7 +196,9 @@ function metal_ore_processing(material, properties, event, id_prefix) {
     // Fallback output checks:
     if (crushed_ore_itemStack.isEmpty()) {
         crushed_ore_itemStack = Item.of(Ingredient.of(`#create:crushed_ores/${material}`).getItemIds()[0]);
-        crushed_ore_itemStack = (crushed_ore_itemStack.isEmpty() || crushed_ore_itemStack.getId() == "minecraft:barrier")? null: crushed_ore_itemStack;
+        if (crushed_ore_itemStack.isEmpty() || crushed_ore_itemStack.getId() == "minecraft:barrier") {
```
Comment (NielsPilgaard) [inline kubejs/server_scripts/normal/unification/unify_ore_processing.js:199]: There were several similar issues in this file below this I'm afraid 😄

Comment (Kanzaji) [reply kubejs/server_scripts/normal/unification/unify_ore_processing.js:199]: Woops! Time to check the entire file for those then 😅

File: kubejs/server_scripts/base/tags/items/minecraft/piglin_loved.js
Comment (Kanzaji) [reply kubejs/server_scripts/base/tags/items/minecraft/piglin_loved.js:1]: I completely forgot that it was already in the release 😅

File: kubejs/server_scripts/base/tags/items/emendatusenigmatica/gold.js [hunk 10]
```diff
@@ -0,0 +1 @@
+// TODO: Delete next breaking update
```
Comment (NielsPilgaard) [inline kubejs/server_scripts/base/tags/items/emendatusenigmatica/gold.js:1]: oh I'm wrong, this file is actually new :P please delete. Sorry!

**Review by NielsPilgaard — APPROVED** [review]

**Comment by NielsPilgaard** [comment]
LGTM, good work! 😃 Wanna give this a once-over @MuteTiefling before I merge?

File: kubejs/server_scripts/base/tags/items/emendatusenigmatica/gold.js (hunk 10 above)
Comment (Kanzaji) [reply kubejs/server_scripts/base/tags/items/emendatusenigmatica/gold.js]: No no no! I was wrong, it is in the release 😅 I just had it modified from the start iirc and forgot that it also was in the previous PR.
obraz

Comment (NielsPilgaard) [reply kubejs/server_scripts/base/tags/items/emendatusenigmatica/gold.js]: Oh, I misinterpreted GitHub 😅

**Comment by MuteTiefling** [comment]
@NielsPilgaard Looks good to me too!

**Review by MuteTiefling — APPROVED** [review]

**Pull request merged**